### DISCUSSION TITLE
Parenthesis if in batch

### DISCUSSION THREAD

**Discussion by Tosyk** [ImageMagick/ImageMagick#6816] [Help] [answered]
### ImageMagick version

7.0.11

### Operating system, version and so on

Windows 11 x64

### Description

Hi,

I can't solve parenthesis here:

```
<code>

if %al% == srgba (
	"%NPATH%\magick.exe" ^
		%1 ^
		-channel rgba ^
		-separate +channel ^
		-swap 1,2 +delete ^
		-combine ^
		-channel B ^
		-evaluate set 100% ^
		+channel ^
		\( %ao% ^
		-channel G -separate +channel ^
		-interpolate Nearest -filter point ^
		-gamma 0.8 ^
		-resize %w%x%h% ^
		\) ^
		-compose CopyBlue -composite ^
		-compose CopyOpacity -composite ^
		-colorspace sRGB ^
		%path%%resname%_RMAM.%EXT%
)

<code>
```

so I get:
```
-compose was unexpected at this time.
```
Can you guys help me?

**Comment by snibgo** [comment, marked as answer]
Backslash `\` is the bash shell escape character. You use the Windows CMD shell, so use caret `^` instead.

**Reply by Tosyk** [reply]
like this?
```
<code>

if %al% == srgba (
	"%NPATH%\magick.exe" ^
		%1 ^
		-channel rgba ^
		-separate +channel ^
		-swap 1,2 +delete ^
		-combine ^
		-channel B ^
		-evaluate set 100% ^
		+channel ^
		^( %ao% ^
		-channel G -separate +channel ^
		-interpolate Nearest -filter point ^
		-gamma 0.8 ^
		-resize %w%x%h% ^
		^) ^
		-compose CopyBlue -composite ^
		-compose CopyOpacity -composite ^
		-colorspace sRGB ^
		%path%%resname%_RMAM.%EXT%
)

<code>
```

**Reply by fmw42** [reply]
Remove the \ in front of the parentheses. That is Unix syntax.  Window syntax does not use \.  If you need to escape the parentheses use ^ before them.  A Windows user can help you further or correct my comments.

**Reply by Tosyk** [reply]
it worked but if I have few children it doesn't work, can you help?

```
if %al% == srgb (
	"%NPATH%\magick.exe" ^
		%1 ^
		-channel rgba ^
		-separate +channel ^
		-swap 1,2 +delete ^
		-combine ^
		-channel B ^
		-evaluate set 100% ^
		+channel ^
		^( %ao% ^
		-channel G -separate +channel ^
		-interpolate Nearest -filter point ^
		-gamma 0.8 ^
		-resize %w%x%h% ^
		^) ^
		-compose CopyBlue -composite ^
		+channel ^
		^( %co% ^
		-channel A -separate +channel ^
		-interpolate Nearest -filter point ^
		-resize %w%x%h% ^
		^) ^
		-compose CopyOpacity -composite ^
		-colorspace sRGB ^
		%path%%resname%_RMAM.%EXT%
)
```

**Reply by fmw42** [reply]
I will leave that to a Windows user.

**Reply by Tosyk** [reply]
I was looking for the answer the whole day. Couldn't find, so I thought it would be wise to ask here

**Reply by jaffamuffin** [reply]
what is the error?

i ran your code here and it produced an output with no errors.

try 100% changing to 100%%

Or change whole routine to 

if %al% == srgb (
  CALL :srgb %1 %ao% %co% %w% %h% %path%%resname%_RMAM.%EXT%
)


:srgb
set ao=%2
set co=%3 
etc..
insert your code from the if here..

GOTO :EOF

**Reply by fmw42** [reply]
In windows, % need to be escaped to %% as in 100% changed to 100%%

**Reply by snibgo** [reply]
> what is the error?
i ran your code here and it produced an output with no errors.

You have no errors, and ask what the error is? I don't understand.

You show some code. The code after `:srgb` will be executed either once or twice. Perhaps that is not what you want.

**Comment by snibgo** [comment]
> it worked but if I have few children it doesn't work, can you help?

What are "children"?

Do you get an error message?

If you are using this in a BAT script, `100%` needs to be `100%%`, with a doubled percentage.

**Reply by Tosyk** [reply]
I think I managed to solve this. ^ - this helped ; D
